src/clients/Client.cc

```diff
@@ -759,8 +757,14 @@ Client::resumeBodyStorage()
 
     handleMoreAdaptedBodyAvailable();
 
-    if (adaptedBodySource != nullptr && adaptedBodySource->exhausted())
-        endAdaptedBodyConsumption();
+    if (adaptedBodySource != nullptr && adaptedBodySource->exhausted()) {
+        if (receivedWholeAdaptedReply || adaptedReplyAborted) {
+            // noteBodyProductionEnded() or noteBodyProducerAborted() could not endAdaptedBodyConsumption()
+            // while waiting for resumeBodyStorage() to consume buffered adapted body data.
+            endAdaptedBodyConsumption();
+        }
+        // else wait for noteBodyProductionEnded() or noteBodyProducerAborted()
```

Comment:
Let's explain _why_ we have to wait even though we know that no more body bytes are coming.

N.B. This explanation will migrate (probably in a form of debugs() statement) into checkAdaptationCompletion() proposed in another change request.

```suggestion
        // else wait for noteBodyProductionEnded() or noteBodyProducerAborted()
        // because we need to know whether we receivedWholeAdaptedReply
```